backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/questionsubmission/QuestionSubmissionController.java

```diff
@@ -0,0 +1,105 @@
+package pt.ulisboa.tecnico.socialsoftware.tutor.questionsubmission;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.*;
+import pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.TutorException;
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.dto.TopicDto;
+import pt.ulisboa.tecnico.socialsoftware.tutor.questionsubmission.dto.ReviewDto;
+import pt.ulisboa.tecnico.socialsoftware.tutor.questionsubmission.dto.QuestionSubmissionDto;
+import pt.ulisboa.tecnico.socialsoftware.tutor.questionsubmission.dto.UserQuestionSubmissionInfoDto;
+import pt.ulisboa.tecnico.socialsoftware.tutor.user.User;
+
+import javax.validation.Valid;
+import java.security.Principal;
+import java.util.List;
+
+import static pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.ErrorMessage.AUTHENTICATION_ERROR;
+
+@RestController
+public class QuestionSubmissionController {
+    @Autowired
+    private QuestionSubmissionService questionSubmissionService;
+
+    @PostMapping(value = "/submissions/{executionId}")
+    @PreAuthorize("hasRole('ROLE_STUDENT') and hasPermission(#executionId, 'EXECUTION.ACCESS')")
+    public QuestionSubmissionDto createQuestionSubmission(Principal principal, @PathVariable int executionId, @Valid @RequestBody QuestionSubmissionDto questionSubmissionDto) {
+        User user = (User) ((Authentication) principal).getPrincipal();
```

Comment:
This is note necessary, to get the autheticated user, if it is not authenticated it fail in @PreAuthorize, check the other services here for same problem